server/agent/tools/create_answer_tool.py

```diff
@@ -46,13 +49,18 @@ class CreateAnswerTool(CustomTool):
     max_tokens: int = 30000
     max_retry: int = 3
     chat_model: str = None
+    group_name: str = None
+    _channel_layer: Any = PrivateAttr()
 
     def __init__(self,
                  data_set: DataSet,
                  chat_model: str,
+                 group_name: str,
                  **kwargs: Any):
         super().__init__(data_set=data_set, chat_model=chat_model, **kwargs)
         self.encoding = tiktoken.encoding_for_model(chat_model)
+        self.group_name = group_name
+        self._channel_layer = get_channel_layer()
```

Comment:
This shouldn't be here - it's a different level of abstraction. You could just add `conversation` as another parameter for the tool interface - and then rely on the `WebSockerCallbackHandler` to take care of it